tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
+    "compilerOptions": {
+      "types": ["node"],
+      "outDir": "./build",
+      "allowJs": true,
+      "target": "es5"
```

Comment:
There are some config options which would be nice to include that help enforce code quality and reduce really weird errors that can occur, Matt Pocock has got [a really good cheat sheet](https://www.totaltypescript.com/tsconfig-cheat-sheet) - could we utilize that as a starting point? 

Additionally:

- `outDir: "./build"` -> convention in the js/ts community is to use `dist` - it'd be great to align with the convention (if possible) 😄 
- `target: "es5"` -> can we target above es5, or is there a specific requirement locking the compilation to es5? I'm not familiar with the level of browser support the previous code was at